lnwallet/size_test.go

```diff
@@ -0,0 +1,124 @@
+package lnwallet_test
+
+import (
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/roasbeef/btcd/blockchain"
+	"github.com/roasbeef/btcd/chaincfg"
+	"github.com/roasbeef/btcd/txscript"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+// TestTransactionWeightEstimate tests that transaction weight estimates are
+// calculated correctly by comparing against an actual (though invalid)
+// transaction matching the template.
+func TestTransactionWeightEstimate(t *testing.T) {
+	netParams := &chaincfg.MainNetParams
+
+	p2pkhAddr, err := btcutil.NewAddressPubKeyHash(
+		make([]byte, 20), netParams)
+	if err != nil {
+		t.Fatalf("Failed to generate address: %v", err)
+	}
+	p2pkhScript, err := txscript.PayToAddrScript(p2pkhAddr)
+	if err != nil {
+		t.Fatalf("Failed to generate scriptPubKey: %v", err)
+	}
+
+	p2wkhAddr, err := btcutil.NewAddressWitnessPubKeyHash(
+		make([]byte, 20), netParams)
+	if err != nil {
+		t.Fatalf("Failed to generate address: %v", err)
+	}
+	p2wkhScript, err := txscript.PayToAddrScript(p2wkhAddr)
+	if err != nil {
+		t.Fatalf("Failed to generate scriptPubKey: %v", err)
+	}
+
+	p2wshAddr, err := btcutil.NewAddressWitnessScriptHash(
+		make([]byte, 32), netParams)
+	if err != nil {
+		t.Fatalf("Failed to generate address: %v", err)
+	}
+	p2wshScript, err := txscript.PayToAddrScript(p2wshAddr)
+	if err != nil {
+		t.Fatalf("Failed to generate scriptPubKey: %v", err)
+	}
+
+	testCases := []struct {
+		numP2PKHInputs  int
+		numP2WKHInputs  int
+		numP2PKHOutputs int
+		numP2WKHOutputs int
+		numP2WSHOutputs int
+		expectedWeight  int
```

Comment:
Seems as this field isn't currently used within the tests. Instead, it's generate in real-time by constructing an equivalent transaction. In that case, it can safely be removed.

Reply:
Yep, this was leftover from how I first tried writing the tests. Will remove.